File: Java/Hard/WrongAnswer/naming-a-company_2.java
class Solution {
    public long distinctNames(String[] ideas) {
        
        Set<String> set = new HashSet<>(Arrays.asList(ideas));
        long result = 0;
        for (int i = 0; i < ideas.length; i++) {
            for (int j = i + 1; j < ideas.length; j++) {
                char[] chars = ideas[i].toCharArray();
                char temp = chars[0];
                chars[0] = ideas[j].charAt(0);
                String newString = new String(chars);
                if (!set.contains(newString)) {
                    result++;
                }
                chars[0] = temp;
                chars = ideas[j].toCharArray();
                chars[0] = ideas[i].charAt(0);
                newString = new String(chars);
                if (!set.contains(newString)) {
                    result++;
                }
            }
        }
        return result;
    }
}

File: Rust/Hard/CompileError/naming-a-company_2.rs
impl Solution {
    pub fn distinct_names(ideas: Vec<String>) -> i64 {
        
        let mut names = HashSet::new();
        for i in 0..ideas.len() {
            for j in i+1..ideas.len() {
                let mut ideaA = ideas[i].clone();
                let mut ideaB = ideas[j].clone();
                let mut a = ideaA.chars().nth(0).unwrap();
                let mut b = ideaB.chars().nth(0).unwrap();
                ideaA.replace_range(0..1, &b.to_string());
                ideaB.replace_range(0..1, &a.to_string());
                if !ideas.contains(&ideaA) && !ideas.contains(&ideaB) {
                    names.insert(ideaA + " " + &ideaB);
                }
            }
        }
        names.len() as i64
    }
}